search-commons/src/main/java/no/unit/nva/search2/ResourceQuery.java

```diff
@@ -177,13 +175,19 @@ protected void setValue(String key, String value) {
             }
         }
 
+        private void addSortOrderToSortQuery(String value) {
+            query.setQueryValue(SORT, mergeParameters(query.getValue(SORT), value));
+        }
+
         @Override
         protected void applyRulesAfterValidation() {
             // convert page to offset if offset is not set
-            if (isNull(query.getValue(OFFSET)) && nonNull(query.getValue(PAGE))) {
-                var page = Integer.parseInt(query.getValue(PAGE));
-                var perPage = Integer.parseInt(query.getValue(PER_PAGE));
-                query.setQueryValue(OFFSET, String.valueOf(page * perPage));
+            if (nonNull(query.getValue(PAGE))) {
+                if (isNull(query.getValue(OFFSET))) {
+                    var page = Integer.parseInt(query.getValue(PAGE));
+                    var perPage = Integer.parseInt(query.getValue(PER_PAGE));
+                    query.setQueryValue(OFFSET, String.valueOf(page * perPage));
+                }
                 query.removeValue(PAGE);
```

Comment:
bugfix-ish, PAGE must not get passed to SWS gateway and opensearch

Reply:
All the tests run without nested if statement:

```
@Override
        protected void applyRulesAfterValidation() {
            // convert page to offset if offset is not set
            if (nonNull(query.getValue(PAGE)) && isNull(query.getValue(OFFSET))) {
                var page = Integer.parseInt(query.getValue(PAGE));
                var perPage = Integer.parseInt(query.getValue(PER_PAGE));
                query.setQueryValue(OFFSET, String.valueOf(page * perPage));
                query.removeValue(PAGE);
            }
            // TODO check if field is set and has value 'all' then populate with all fields
        }
```

Should it be nested if statement? If yes, it would be nice to have a test for it, in case someone decides to refactor it later.

Reply:
```
URI.create("https://example.com/?category=hello+world&user=12%203&page=2"),
URI.create("https://example.com/?category=hello+world&user=12%203&offset=30"),
URI.create("https://example.com/?category=hello+world&user=12%203&from=30&results=10"),
```
1 test with page
1 test with offset
1 test with from

Reply:
But there is no failing tests when we combine this two if statements, which means that they do not need to be nested?

Reply:
added these -> 
```
 void uriParamsToResourceParams()...
 
        assertNotNull(resourceParameters.getValue(OFFSET));
        assertNull(resourceParameters.getValue(PAGE));
```

Reply:
But should this if statement be nested? Like
```
if() {
   if() {
  }
}
```